src/box/lua/iproto.lua

```diff
@@ -0,0 +1,30 @@
+-- Sets IPROTO request handler callback (second argument) for the given request
+-- type (first argument, number).
+-- Passing nil as the callback resets the corresponding request handler.
+box.iproto.override = function(request_type, callback)
+    local trigger = require('trigger')
+
+    if request_type == nil then
+        box.error(box.error.PROC_LUA,
+                  'Usage: box.iproto.override(request_type, callback)')
+    end
+    if type(request_type) ~= 'number' then
+        box.error(box.error.PROC_LUA,
+                  ("bad argument #1 to 'override' (number expected, " ..
+                   "got %s)"):format(type(callback)))
+    end
+    if callback ~= nil and type(callback) ~= 'function' then
+        box.error(box.error.PROC_LUA,
+                  ("bad argument #2 to 'override' (function expected, " ..
+                   "got %s)"):format(type(callback)))
```

Comment:
I am not an expert on the trigger subsystem, but can't it also be a callable table now? @drewdzzz

Reply:
AFAIK, it's hard to support callable tables here. Let's allow only functions here. Callable objects can be set via `trigger.set()`.

Reply:
We cannot check if an object is callable from Lua (for example, AFAIK, there is no way to check if `cdata` is callable from Lua). But if we really want to accept callable objects, we could call `trigger.set` with `pcall` or rewrite this function in C. But as for me, it's OK to accept only functions here.